src/Observers/EntityObserver.php

```diff
@@ -64,4 +64,31 @@ private function resolveEntityData($entity)
 
         return $data;
     }
+
+    private function removeRelations($entity)
+    {
+        $entityCollection = collect($entity);
+        $databaseProperties = $this->getModelProperties($entity);
+
+        return $entityCollection->only($databaseProperties->toArray());
+    }
+
+    private function resolveUpdateEntityData($entity, $data)
+    {
+        if (method_exists($entity, 'ignoreSyncAttributes')) {
+            foreach ($entity->ignoreSyncAttributes() as $attribute) {
+                unset($data[$attribute]);
+            }
+        }
+
+        return $data;
+    }
+
+    /*
+     * Fetches all the column names from the current database table you are querying
+     */
+    private function getModelProperties($entity)
+    {
+        return collect($entity->getConnection()->getSchemaBuilder()->getColumnListing($entity->getTable()));
```

Comment:
Remove collect() from this method. `getColumnListing()` returns an array and we never use the collection logic, other then fetching the array version we already have.